### ISSUE TITLE
swmm-python fails to compile in Apple M1

### ISSUE THREAD

**Issue by abhiramm7** [pyswmm/swmm-python#93]
Hi all, 

swmm-python fails to compile in M1 Macs. It seems to be an openmp issue. Digging more into it, I've discovered that llvm does not yet fully support openmp (see https://github.com/seq-lang/seq/issues/248#issuecomment-985193820). The same issue seems to persist with gcc as well. 

<img width="1472" alt="Screen Shot 2021-12-29 at 10 47 52 PM" src="https://user-images.githubusercontent.com/14915561/147720154-b7faaa15-a1ed-4295-9687-63e69154d47f.png">

I found two ways to address this issue:

1. Compile with `set(CMAKE_OSX_ARCHITECTURES "x86_64")` 
2. Disable openmp in CmakeLists

Though 1 works, it would need x86_64 python installed. For now I think 2 might be the best option. We would just have to update cmakelists file. 
```cmake
# If wheel build on Apple fetch and build OpenMP Library
if (APPLE AND (NOT CMAKE_OSX_ARCHITECTURES STREQUAL "arm64"))
    include(./extern/openmp.cmake)
else()
    find_package(OpenMP
        OPTIONAL_COMPONENTS
            C
    )
endif()
```

I'll keep looking for a way to support openmp on m1 Macs. For now, I am able to compile swmm-python and run pyswmm without openmp.

**Comment by michaeltryby** [comment]
@abhiramm7 if you refer to homebrew recipe for libomp you can see that they are supporting M1 macs. I would recommend upgrading to libomp v13 like they do. Currently we build v11.1.

**Comment by abhiramm7** [comment]
Oh cool. I am using v13 libomp. May be my compiler might not be seeing the libomp, I'll dig into it.

**Comment by michaeltryby** [comment]
@abhiramm7 swmm-toolkit wheel builds its own version of libomp for binary compatibility with Python. Refer to external/openmp.cmake

**Comment by michaeltryby** [comment]
Deployment target for Python on X86_64 Mac is 10.9. Doesn’t make sense on an M1 Mac though …

**Comment by michaeltryby** [comment]
Helpful discussion

https://discuss.python.org/t/providing-binary-packages-for-apple-m1/9288/5

**Comment by michaeltryby** [comment]
This is actually a significant task involving a port of SWMM to a new processor. I would be wise to establish a new set of benchmarks for the SWMM build

**Comment by karosc** [comment]
this is resolved by [SWMM #403](https://github.com/pyswmm/Stormwater-Management-Model/pull/403) and [swmm-toolkit #122](https://github.com/pyswmm/swmm-python/pull/122)
